packages/service-library/src/servicelib/rabbitmq/_client.py

```diff
@@ -20,6 +21,63 @@
 
 _DEFAULT_PREFETCH_VALUE: Final[int] = 10
 _DEFAULT_RABBITMQ_EXECUTION_TIMEOUT_S: Final[int] = 5
+_HEADER_X_DEATH: Final[str] = "x-death"
+
+DEFAULT_UNEXPECTED_ERROR_RETRY_DELAY_S: Final[float] = 1
+DEFAULT_UNEXPECTED_ERROR_MAX_ATTEMPTS: Final[NonNegativeInt] = 15
+
+
+def _get_delayed_exchange_name(exchange_name: str) -> str:
+    return f"delayed_{exchange_name}"
+
+
+def _get_x_death_count(message: aio_pika.abc.AbstractIncomingMessage) -> int:
+    count: int = 0
+
+    x_death: list[dict[str, Any]] = message.headers.get(_HEADER_X_DEATH, [])
+    if x_death:
+        count = x_death[0]["count"]
+
+    return count
+
+
+async def _on_message(
+    message_handler: MessageHandler,
+    max_retries_upon_error: int,
+    message: aio_pika.abc.AbstractIncomingMessage,
+) -> None:
+    async with message.process(requeue=True, ignore_processed=True):
+        count = _get_x_death_count(message)
```

Comment:
This count you need in case of exception, righ? Since you have to compute for every message, why not to put it in the exception handler (e.g. before the warning) this way is more clear

Reply:
good one, forgot it there form a previous implementation.